driver/pipeline/acquisition.cpp

```diff
@@ -21,30 +23,24 @@ Acquisition::Acquisition(
     WriterConfig writer_config,
     std::shared_ptr<Source> source,
```

Comment:
Why does this need to be a shared pointer?

Reply:
Because in the case of NI for example, we have the DigitalWriteSink which has a member StateSource which it needs to update the state of the digital outputs, but it also needs to be accessed through the acquisition pipe to read out those states.